certbot/main.py

```diff
@@ -911,8 +912,11 @@ def enhance(config, plugins):
     if not config.chain_path:
         lineage = cert_manager.lineage_for_certname(config, config.certname)
         config.chain_path = lineage.chain_path
-    le_client = _init_le_client(config, authenticator=None, installer=installer)
-    le_client.enhance_config(domains, config.chain_path, ask_redirect=False)
+    if oldstyle_enh:
+        le_client = _init_le_client(config, authenticator=None, installer=installer)
+        le_client.enhance_config(domains, config.chain_path, ask_redirect=False)
+    if enhancements.is_supported(config):
+        enhancements.enable(lineage, domains, installer, config)
```

Comment:
This enhancement can currently only be enabled with the `enhance` subcommand. I think in the long term we also want it to work for `run` and `install`.

If this was on purpose and you want to do it later, that's fine, otherwise, I think this should be moved to `le_client.enhance_config`.

Reply:
I will add this functionality after #6097 lands.